src/eduid_userdb/actions/db.py

```diff
@@ -51,6 +52,7 @@ def __init__(self, db_uri, db_name='eduid_actions', collection='actions'):
         super(ActionDB, self).__init__(db_uri, db_name, collection)
 
         self._cache = {}
+        self._left_in_cache = {}
```

Comment:
self._cache[cachekey] is a [mongo cursor](http://api.mongodb.org/python/current/api/pymongo/cursor.html), that can be rewind, etc. It has a count() method that always returns the number of objects found; it has a retrieved property that has very similar semantics to count()... It does not make the number of objects left from the current position available; I spent quite a while this morning on this issue. The documentation doesn't mention a __len__ method that will do just that, and it would be a surprise if it existed.

However, now that we discuss it, a simpler solution occurs to me, which is to iterate over the cursor on first read and cache a list with those results. I'll do that and see if you like it better.

Another solution of course is to remove the cache altogether.